### PR TITLE
Allow downloading logs

### PR DESCRIPTION
**Pull request by balloob**
Add a download logs button to all dialogs that show logs (both webserial and on the server) or process output (install, validate, update all etc)

image

### CODE REVIEW

File: src/components/remote-process.ts [hunk 1]
```diff
@@ -11,6 +11,10 @@ class ESPHomeRemoteProcess extends HTMLElement {
   private _abortController?: AbortController;
   private _setup = false;
 
+  public logs(): string {
+    return this.shadowRoot!.querySelector("div")!.innerText;
```
[the remark this answers is not in the document]
Comment (balloob) [reply src/components/remote-process.ts]: Fixed

File: src/logs-webserial/ewt-console.ts [hunk 2]
```diff
@@ -11,6 +11,10 @@ export class EwtConsole extends HTMLElement {
   private _console?: ColoredConsole;
   private _cancelConnection?: () => Promise<void>;
 
+  public logs(): string {
```
[the remark this answers is not in the document]
Comment (balloob) [reply src/logs-webserial/ewt-console.ts:14]: Fixed

**Comment by balloob** [comment]
Thanks for the comments. Addressed.

Btw this is not Polymer but Lit. Besides the rendering of HTML and attaching event listeners it works just like VanillaJS, you use the same methods to query DOM and interact with elements etc.

**Comment by nagyrobi** [comment]
Finally, thanks!

**Review by jesserockz — APPROVED** [review]

**Pull request merged**